ledger/state_test.go

```diff
@@ -0,0 +1,464 @@
+		Cbor:     []byte{0x82, 0x00, 0x81, 0x00}, // minimal valid CBOR
+		Type:     5,                               // Byron
+	}
+
+	// Add block to chain
+	primaryChain := cm.PrimaryChain()
+	if err := primaryChain.AddBlock(testBlock, nil); err != nil {
+		t.Fatalf("failed to add test block: %v", err)
+	}
+
+	// Create LedgerState
+	ls := &LedgerState{
+		chain: primaryChain,
+		db:    db,
+	}
+
+	// Test GetBlock with valid point
+	point := ocommon.NewPoint(testBlock.Slot, testBlock.Hash)
+	result, err := ls.GetBlock(point)
+	if err != nil {
+		t.Fatalf("GetBlock returned unexpected error: %v", err)
+	}
+
+	// Verify returned block matches
+	if result.Slot != testBlock.Slot {
+		t.Errorf("Expected slot %d, got %d", testBlock.Slot, result.Slot)
+	}
+	if string(result.Hash) != string(testBlock.Hash) {
+		t.Errorf("Expected hash %x, got %x", testBlock.Hash, result.Hash)
+	}
+	if result.Number != testBlock.Number {
+		t.Errorf("Expected block number %d, got %d", testBlock.Number, result.Number)
+	}
+}
+
+// TestGetBlock_NotFound tests that GetBlock returns an error when block not found
+func TestGetBlock_NotFound(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create LedgerState
+	ls := &LedgerState{
+		chain: cm.PrimaryChain(),
+		db:    db,
+	}
+
+	// Test GetBlock with non-existent point
+	point := ocommon.NewPoint(99999, []byte{0xff, 0xff, 0xff, 0xff})
+	result, err := ls.GetBlock(point)
+	if err == nil {
+		t.Fatal("Expected error for non-existent block, got nil")
+	}
+
+	// Verify zero value is returned on error
+	if result.Slot != 0 {
+		t.Errorf("Expected zero slot on error, got %d", result.Slot)
+	}
+	if len(result.Hash) != 0 {
+		t.Errorf("Expected empty hash on error, got %x", result.Hash)
+	}
+}
+
+// TestGetBlock_ReturnsValueNotPointer tests that GetBlock returns value type, not pointer
+func TestGetBlock_ReturnsValueNotPointer(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create test block
+	testBlock := models.Block{
+		Hash:     []byte{0xaa, 0xbb, 0xcc, 0xdd},
+		Slot:     54321,
+		Number:   200,
+		PrevHash: []byte{0x00, 0x00, 0x00, 0x00},
+		Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+		Type:     5,
+	}
+
+	// Add block to chain
+	primaryChain := cm.PrimaryChain()
+	if err := primaryChain.AddBlock(testBlock, nil); err != nil {
```

Comment:
@coderabbitai This should be a "github.com/blinklabs-io/gouroboros/ledger/common".Block not a models.Block struct. You can use Decode() on models.Block to get a ledger.Block suitable for use here.